modules/packages/msiexec.bat

```diff
@@ -96,8 +96,10 @@ rem Install this file if it exists
     goto :EOF
   )
 
-  %MSIEXEC% /quiet /passive /qn /norestart /i %1
-  rem TODO options, error checking
+  %MSIEXEC% /quiet /passive /qn /norestart /l*v %1_cfengine_install.log /i %1
+  if not errorlevel 0 (
+    echo ErrorMessage=msiexec ErrorLevel was %ErrorLevel% for file %1 logfile at %1_cfengine_install.log
+  )

```

Comment:
might be a good idea to put
```
) else (
  del %1_cfengine_install.log
)
```

To remove the log file in case of success.

Reply:
What is %1? the package name? So a log file for each named package that is failing? Log file is over-written or appended to if it fails over multiple agent runs?

Reply:
overwritten

Reply:
So I have changed this slightly by putting the files in a dir so that log_rotation.cf will clean things up here.

Each unique msi file will have it's own log file and on repeated attempts it will be overwritten.

In most cases I would imagine this will be just fine as likely the errors will be the same over and over for a specific msi.

Reply:
Ack

Reply:
> might be a good idea to put
> 
> ```
> ) else (
>   del %1_cfengine_install.log
> )
> ```
> 
> To remove the log file in case of success.

@nickanderson actually I'm wondering if just keeping the success log isn't a good idea as well, in case `msiexec.exe` returns success but the install actually doesn't quite work for some reason, then you have access to the detailed log!